src/LiquidityPool.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+
+import "../src/MyToken.sol";
+
+contract LiquidityPool {
+    MyToken public tokenA;
+    MyToken public tokenB;
+
+    mapping(address => uint256) public balancesA;
+    mapping(address => uint256) public balancesB;
+
+    uint256 public exchangeRate; // Storage variable for the exchange rate
+
+    constructor(MyToken _tokenA, MyToken _tokenB, uint256 initialRate) {
+        tokenA = _tokenA;
+        tokenB = _tokenB;
+        exchangeRate = initialRate; 
+    }
+
+    // Update the exchange rate based on the constant product formula
+    function updateExchangeRate() internal {
+        require(balancesA[msg.sender] > 0 && balancesB[msg.sender] > 0, "Insufficient liquidity");
+
+        exchangeRate = balancesB[msg.sender] / balancesA[msg.sender];
```

Comment:
`exchangeRate` should not be updated according to the balances of the user `msg.sender`. it should be set using the balances of tokens A and B